lib/CSkillHandler.cpp

```diff
@@ -172,6 +201,8 @@ CSkill * CSkillHandler::loadFromJson(const JsonNode & json, const std::string &
 		// parse skill description - tracked separately
 		if(vstd::contains(levelNode.Struct(), "description") && !levelNode["description"].isNull())
 			skill->setDescription(levelNode["description"].String(), level);
+		else
+			skill->setDescription(skillInfo(skill->id, level), level);
```

Comment:
Not really sure why this is needed. I would have expected legacy data to get json-merged with skills.json, so that legacy descriptions should be part of the JsonNode passed to loadFromJson. Does not appear to happen though.

Reply:
You need "index" field to merge legacy data.

Reply:
Ah, I see, missed that. Thanks.